internal/test_cases/single_line_exact_match_output_test_case.go

```diff
@@ -0,0 +1,64 @@
+package test_cases
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+	"github.com/fatih/color"
+)
+
+// SingleLineExactMatchTestCase internally creates a SingleLineOutputTestCase with a validator that matches the output against a pattern.
+// We look for an exact match in this case, so our error logs can take advantage of color to highlight the exact mismatch.
+type SingleLineExactMatchTestCase struct {
+	// Command is the command to execute, whose output will be matched against ExpectedPattern.
+	Command string
+
+	// ExpectedPattern is the regex that is evaluated against the command's output.
+	ExpectedPattern string
```

Comment:
@ryan-gang for an exact match, why do we need a pattern? Can't we just pass a string?